frontend/src/metabase/visualizations/echarts/cartesian/chart-measurements/index.ts

```diff
@@ -187,32 +229,52 @@ export const getTicksDimensions = (
       ) + CHART_STYLE.axisTicksMarginY;
   }
 
+  const currentBoundaryWidth =
+    chartWidth -
+    CHART_STYLE.padding.x * 2 -
+    ticksDimensions.yTicksWidthLeft -
+    ticksDimensions.yTicksWidthRight;
+
+  ticksDimensions.minXTickSpacing = getMinXTickSpacing(
+    chartModel.dataset,
+    settings,
+    currentBoundaryWidth,
+    chartModel.xAxisModel.formatter,
+    renderingContext,
+  );
+
   const isTimeSeries = isTimeSeriesAxis(chartModel.xAxisModel);
-  const hasBottomAxis = !!settings["graph.x_axis.axis_enabled"];
+  let axisEnabledSetting = settings["graph.x_axis.axis_enabled"];
+  const hasBottomAxis = !!axisEnabledSetting;
+
   if (hasBottomAxis) {
-    ticksDimensions.xTicksHeight =
-      getXAxisTicksHeight(
-        chartModel.dataset,
+    const { firstXTickWidth, lastXTickWidth, maxXTickWidth } =
+      getXAxisTicksWidth(
+        chartModel.transformedDataset,
         settings,
         chartModel.xAxisModel.formatter,
         renderingContext,
-      ) +
+      );
+    ticksDimensions.firstXTickWidth = firstXTickWidth;
+    ticksDimensions.lastXTickWidth = lastXTickWidth;
+    ticksDimensions.maxXTickWidth = maxXTickWidth;
+
+    axisEnabledSetting = getAutoAxisEnabledSetting(
```

Comment:
I think we need to compute this before `firstXTickWidth` and  `lastXTickWidth`. For example, if `axisEnabledSetting` is auto (true) then we assume ticks are not rotated and the width can be substantial. So if we later change `axisEnabledSetting` to 90 degrees, it will produce incorrect paddings

Reply:
Done